src/VisualStudio/Core/Impl/SolutionExplorer/SourceGeneratedFileItems/SourceGeneratedFileItem.cs

```diff
@@ -51,29 +51,37 @@ public override object GetBrowseObject()
             return new BrowseObject(this);
         }
 
-        public override IInvocationController InvocationController => InvocationControllerImpl.Instance;
+        public override IInvocationController InvocationController
+            => new InvocationControllerImpl(_threadingContext);
 
         private sealed class InvocationControllerImpl : IInvocationController
         {
-            public static IInvocationController Instance = new InvocationControllerImpl();
+            private readonly IThreadingContext _threadingContext;
 
-            public bool Invoke(IEnumerable<object> items, InputSource inputSource, bool preview)
+            public InvocationControllerImpl(IThreadingContext threadingContext)
             {
-                var didNavigate = false;
+                _threadingContext = threadingContext;
+            }
 
-                foreach (var item in items.OfType<SourceGeneratedFileItem>())
+            public bool Invoke(IEnumerable<object> items, InputSource inputSource, bool preview)
+            {
+                return _threadingContext.JoinableTaskFactory.Run(async () =>
                 {
-                    var documentNavigationService = item.Workspace.Services.GetService<IDocumentNavigationService>();
-                    if (documentNavigationService != null)
+                    var didNavigate = false;
+                    foreach (var item in items.OfType<SourceGeneratedFileItem>())
                     {
-                        // TODO: we're navigating back to the top of the file, do we have a way to just bring it to the focus and that's it?
-                        // TODO: Use a threaded-wait-dialog here so we can cancel navigation.
-                        didNavigate |= item._threadingContext.JoinableTaskFactory.Run(() =>
-                            documentNavigationService.TryNavigateToPositionAsync(item.Workspace, item.DocumentId, position: 0, CancellationToken.None));
+                        var documentNavigationService = item.Workspace.Services.GetService<IDocumentNavigationService>();
+                        if (documentNavigationService != null)
+                        {
+                            // TODO: we're navigating back to the top of the file, do we have a way to just bring it to the focus and that's it?
+                            // TODO: Use a threaded-wait-dialog here so we can cancel navigation.
+                            didNavigate |= await documentNavigationService.TryNavigateToPositionAsync(
+                                _threadingContext, item.Workspace, item.DocumentId, position: 0, CancellationToken.None).ConfigureAwait(false);
```

Comment:
Before you were using item._threadingContext, did you not want to keep using that?

Reply:
i likely missed that.  it's probably fine either way, but i don't mind switching back.